### PR TITLE
Add instructions for switching to/from MUXControlsInnerLoop.sln

### LINKED ISSUE #2766: MUXControlsInnerLoop solution fails to build
**Describe the bug**
The MUXControls solution builds successfully, and the test app deploys and runs properly. However, the MUXControlsInnerLoop solution fails to build with the following errors:
```
1>\microsoft-ui-xaml\BuildOutput\Debug\x64\Microsoft.UI.Xaml\XamlMetadataProviderGenerated.cpp(424,1): error C2872: 'ElementFactoryGetArgs': ambiguous symbol
1>\microsoft-ui-xaml\BuildOutput\Debug\x64\Microsoft.UI.Xaml\XamlMetadataProviderGenerated.cpp(425,1): error C2872: 'ElementFactoryGetArgs': ambiguous symbol
1>\microsoft-ui-xaml\BuildOutput\Debug\x64\Microsoft.UI.Xaml\XamlMetadataProviderGenerated.cpp(432,1): error C2872: 'ElementFactoryGetArgs': ambiguous symbol
1>\microsoft-ui-xaml\BuildOutput\Debug\x64\Microsoft.UI.Xaml\XamlMetadataProviderGenerated.cpp(433,1): error C2872: 'ElementFactoryGetArgs': ambiguous symbol
1>\microsoft-ui-xaml\BuildOutput\Debug\x64\Microsoft.UI.Xaml\XamlMetadataProviderGenerated.cpp(461,1): error C2872: 'ElementFactoryRecycleArgs': ambiguous symbol
1>\microsoft-ui-xaml\BuildOutput\Debug\x64\Microsoft.UI.Xaml\XamlMetadataProviderGenerated.cpp(462,1): error C2872: 'ElementFactoryRecycleArgs': ambiguous symbol
1>\microsoft-ui-xaml\BuildOutput\Debug\x64\Microsoft.UI.Xaml\XamlMetadataProviderGenerated.cpp(469,1): error C2872: 'ElementFactoryRecycleArgs': ambiguous symbol
1>\microsoft-ui-xaml\BuildOutput\Debug\x64\Microsoft.UI.Xaml\XamlMetadataProviderGenerated.cpp(470,1): error C2872: 'ElementFactoryRecycleArgs': ambiguous symbol
```

**Steps to reproduce the bug**
1. Clone this repo (commit 865e4fcc00e8649baeaec1ba7daeca398671aa72)
2. Open the MUXControls solution and build it.
3. Deploy the MUXControlsTestApp project.
4. Open the MUXControlsInnerLoop solution and build it. Observe that it fails.

**Expected behavior**
MUXControlsInnerLoop should build successfully on a fresh clone if the MUXControls solution builds successfully.

**Version Info**
Commit 865e4fcc00e8649baeaec1ba7daeca398671aa72

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2020 Update (19041)            | Yes   |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               | N/A |
| Surface Hub        | N/A |
| IoT                | N/A |

**Additional context**

### PR DESCRIPTION
**Pull request by yoshiask**
## Description
Adds a short paragraph explaining that the BuildOutput folder needs to be deleted when switching between the MUXControls.sln and MUXControlsInnerLoop.sln.

## Motivation and Context
Clear some confusion newcomers to the repo might have when attempting to build the solutions.
Closes #2766

### CODE REVIEW

File: docs/developer_guide.md [hunk 1]
```diff
@@ -53,6 +53,9 @@ A full list of all areas can be found in the **FeatureArea.props** files. To inc
 <FeatureRepeaterEnabled>productOnly</FeatureRepeaterEnabled>
 ```
 
+When switching between the **MUXControls.sln** and **MUXControlsInnerLoop.sln**, make sure you either delete the **BuildOutput** folder in the solution's root directory or run `git clean -dfx`.
+Note that `git clean -dfx` deletes all untracked files, so it may also delete any changes you made. To see what would be deleted by that command without actually deleting anything, run `git clean -dfxn`.
+
```
Comment (chingucoding) [inline docs/developer_guide.md:58]: Maybe we should also point out why this is needed (i.e. that it get's rid of files that MSBUILD clean does not remove)?

**Comment by kmahone** [comment]
Thanks @yoshiask for making this change.

File: docs/developer_guide.md (hunk 1 above)
Comment (StephenLPeters) [reply docs/developer_guide.md:58]: @yoshiask I think this is a good suggestion :)

**Comment by yoshiask** [comment]
I've added my first attempt at explaining why, hopefully I portrayed the cause well enough.

**Comment by chingucoding** [comment]
I think the real issue is that msbuild doesn't clean the files properly. A lot of files do get cleaned when building the solution, so sharing a folder is not necessarily the problem. The issue is that msbuild leaves some files there which are incompatible, that's why you need to purge that folder manually.

**Pull request merged**